libraw/src/main/cpp/anrdroidraw.cpp

```diff
@@ -74,6 +74,63 @@ extern "C" JNIEXPORT jint JNICALL Java_com_homesoft_photo_libraw_LibRaw_getRight
 extern "C" JNIEXPORT jint JNICALL Java_com_homesoft_photo_libraw_LibRaw_getOrientation(JNIEnv* env, jobject jLibRaw){
     return getLibRaw(env, jLibRaw)->imgdata.sizes.flip;
 }
+
+extern "C" JNIEXPORT jfloatArray JNICALL Java_com_homesoft_photo_libraw_LibRaw_getCameraMul(JNIEnv* env, jobject jLibRaw) {
+    auto floatArray = env->NewFloatArray(4);
+    auto cam_mul = getLibRaw(env, jLibRaw)->imgdata.color.cam_mul;
+    env->SetFloatArrayRegion(floatArray, 0, 4, cam_mul);
+    return floatArray;
+}
+
+extern "C" JNIEXPORT jobjectArray JNICALL Java_com_homesoft_photo_libraw_LibRaw_getWhiteBalanceCoefficientsWithTemps(JNIEnv* env, jobject jLibRaw) {
+    auto libRaw = getLibRaw(env, jLibRaw);
+    int elementSize = sizeof(libRaw->imgdata.color.WBCT_Coeffs[0][0]);
+    int firstRowSize = sizeof(libRaw->imgdata.color.WBCT_Coeffs[0]) / elementSize;
+    const int arraySize = sizeof(libRaw->imgdata.color.WBCT_Coeffs) / elementSize / firstRowSize;
+    const int subarraySize = firstRowSize;
+    auto initial = env->NewFloatArray(subarraySize);
+
+    auto floatArrayClass = env->FindClass("[F");
+    if (floatArrayClass == nullptr) {
+        return nullptr;
+    }
+    auto outerArray = env->NewObjectArray(arraySize, floatArrayClass, initial);
+    for (int i=0; i < arraySize; i++) {
+        auto floatArray = env->NewFloatArray(subarraySize);
+        // color temp, r, g, b, g1
+        env->SetFloatArrayRegion(floatArray, 0, subarraySize, libRaw->imgdata.color.WBCT_Coeffs[i]);
+        env->SetObjectArrayElement(outerArray, i, floatArray);
+        env->DeleteLocalRef(floatArray);
+    }
+
+    return outerArray;
+}
+
+extern "C" JNIEXPORT jobjectArray JNICALL Java_com_homesoft_photo_libraw_LibRaw_getWhiteBalanceCoefficients(JNIEnv* env, jobject jLibRaw) {
+    auto libRaw = getLibRaw(env, jLibRaw);
+    int elementSize = sizeof(libRaw->imgdata.color.WB_Coeffs[0][0]);
+    int firstRowSize = sizeof(libRaw->imgdata.color.WB_Coeffs[0]) / elementSize;
+    const int arraySize = sizeof(libRaw->imgdata.color.WB_Coeffs) / firstRowSize / elementSize;
+    const int subarraySize = firstRowSize;
+    auto initial = env->NewIntArray(subarraySize);
+
+    auto intArrayClass = env->FindClass("[I");
+    if (intArrayClass == nullptr) {
```

Comment:
I feel this check is overkill.  If you can't find int[], you are in trouble.  Same for float[] above.

Reply:
reasonable! just saw it done elsewhere, but functionally you're absolutely right. will remove.